src/Features/Core/Portable/Completion/CompletionProvider.cs

```diff
@@ -69,11 +69,20 @@ public virtual Task<CompletionDescription> GetDescriptionAsync(Document document
         /// <param name="item">The item to be committed.</param>
         /// <param name="commitKey">The optional key character that caused the commit.</param>
         /// <param name="cancellationToken"></param>
-        public virtual Task<CompletionChange> GetChangeAsync(
-            Document document, CompletionItem item, char? commitKey, CancellationToken cancellationToken)
-        {
-            return Task.FromResult(CompletionChange.Create(new TextChange(item.Span, item.DisplayText)));
-        }
+        public virtual Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, char? commitKey, CancellationToken cancellationToken)
+            => Task.FromResult(CompletionChange.Create(new TextChange(item.Span, item.DisplayText)));
+
+        /// <summary>
+        /// Gets the change to be applied when the specified item is committed. Change is
+        /// generated via a custom option set instead of using the document's options.
+        /// </summary>
+        /// <param name="document">The current document.</param>
+        /// <param name="optionSet">The set of options to use to generate the change.</param>
+        /// <param name="item">The item to be committed.</param>
+        /// <param name="commitKey">The optional key character that caused the commit.</param>
+        /// <param name="cancellationToken"></param>
+        internal virtual Task<CompletionChange> GetChangeAsync(Document document, OptionSet optionSet, CompletionItem item, char? commitKey, CancellationToken cancellationToken)
+            => GetChangeAsync(document, item, commitKey, cancellationToken);
```

Comment:
@333fred FYI about the new internal `GetChangeAsync` overload.

Reply:
Adding a new overload concerns me. We only just removed the last one with https://github.com/dotnet/roslyn/pull/53261, and this has the same opportunity for us to break the public API version again.

Reply:
I removed the overload and took a different approach, so I don't think this should be a concern anymore